pkg/api/api.go

```diff
@@ -290,7 +290,7 @@ func (a *API) RegisterQuerier(queryable storage.Queryable, engine *promql.Engine
 	cacheGenHeaderMiddleware := getHTTPCacheGenNumberHeaderSetterMiddleware(tombstonesLoader)
 	promHandler := fakeRemoteAddr(cacheGenHeaderMiddleware.Wrap(promRouter))
 
-	a.registerRouteWithRouter(router, a.cfg.PrometheusHTTPPrefix+"/api/v1/read", querier.RemoteReadHandler(queryable), true, "GET")
+	a.registerRouteWithRouter(router, a.cfg.PrometheusHTTPPrefix+"/api/v1/read", querier.RemoteReadHandler(queryable), true, "POST")
```

Comment:
is there any harm in supporting both `GET` and `POST`?

Reply:
`GET` will always return an error since the handler will always try to decode an empty response body. If we only specify `POST`, `GET` will return a method no valid which is what we would want in this case.

Reply:
How has this ever worked?

Reply:
We used to not do any form of method validation.